src/mds/CDir.cc

```diff
@@ -3061,13 +3081,17 @@ void CDir::scrub_maybe_delete_info()
 bool CDir::scrub_local()
 {
   assert(is_complete());
-  bool rval = check_rstats();
+  bool rval = check_rstats(true);
 
   if (rval) {
     scrub_info();
     scrub_infop->last_local.time = ceph_clock_now(g_ceph_context);
     scrub_infop->last_local.version = get_projected_version();
     scrub_infop->last_scrub_dirty = true;
+  } else if (inode->scrub_infop &&
+	     inode->scrub_infop->header &&
+	     inode->scrub_infop->header->repair) {
+    cache->repair_dirfrag_stats(this, NULL);
```

Comment:
So we've got this repair function doing real work that can fail (ESTALE), but meanwhile all the callers of scrub_local() are now ignoring its return value. Which means that rstat failures in scrubbing just get lost and never reported. We need to report them somehow (put it in central log and switch to HEALTH_WARN?) while the stats are bad, until they're fixed. (Which may also require some tricky shenanigans when migrations actually happen...not sure how to best handle that. Maybe we can punt on actually verifying stuff gets updated correctly, until multi-MDS is more stable anyway.)